src/Sylius/Behat/Context/Api/Admin/ManagingCountriesContext.php

```diff
@@ -16,58 +16,63 @@
 use ApiPlatform\Core\Api\IriConverterInterface;
 use Behat\Behat\Context\Context;
 use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Behat\Client\RequestBuilder;
+use Sylius\Behat\Client\RequestBuilderFactoryInterface;
+use Sylius\Behat\Client\RequestFactoryInterface;
 use Sylius\Behat\Client\ResponseCheckerInterface;
 use Sylius\Behat\Context\Api\Resources;
 use Sylius\Behat\Service\SharedStorageInterface;
+use Sylius\Bundle\ApiBundle\Provider\PathPrefixes;
 use Sylius\Component\Addressing\Model\CountryInterface;
 use Sylius\Component\Addressing\Model\ProvinceInterface;
 use Symfony\Component\Intl\Countries;
 use Webmozart\Assert\Assert;
 
 final class ManagingCountriesContext implements Context
 {
-    private ApiClientInterface $client;
-
-    private ResponseCheckerInterface $responseChecker;
-
-    private SharedStorageInterface $sharedStorage;
-
-    private IriConverterInterface $iriConverter;
-
     public function __construct(
-        ApiClientInterface $client,
-        ResponseCheckerInterface $responseChecker,
-        SharedStorageInterface $sharedStorage,
-        IriConverterInterface $iriConverter
+        private ApiClientInterface $client,
+        private ResponseCheckerInterface $responseChecker,
+        private SharedStorageInterface $sharedStorage,
+        private IriConverterInterface $iriConverter,
+        private RequestFactoryInterface $requestFactory,
+        private RequestBuilderFactoryInterface $requestBuilderFactory,
     ) {
-        $this->client = $client;
-        $this->responseChecker = $responseChecker;
-        $this->sharedStorage = $sharedStorage;
-        $this->iriConverter = $iriConverter;
     }
 
     /**
      * @When I want to add a new country
      */
     public function iWantToAddANewCountry(): void
     {
-        $this->client->buildCreateRequest(Resources::COUNTRIES);
+        $requestBuilder = $this->requestBuilderFactory->create(
+            PathPrefixes::ADMIN_PREFIX,
+            Resources::COUNTRIES,
+            'Authorization',
+            $this->sharedStorage->get('token')
+        );
+        $this->sharedStorage->set('request-builder', $requestBuilder);
```

Comment:
```suggestion
        $this->sharedStorage->set('request_builder', $requestBuilder);
```
and everywhere else, as we are using snake_case for keys in shared storage